lib/hardwarekey/agent.go

```diff
@@ -91,7 +95,7 @@ func NewAgentServer(ctx context.Context, s hardwarekey.Service, keyAgentDir stri
 		return nil, trace.Wrap(err)
 	}
 
-	grpcServer := hardwarekeyagent.NewServer(ctx, s, credentials.NewServerTLSFromCert(&cert))
+	grpcServer := hardwarekeyagent.NewServer(ctx, s, credentials.NewServerTLSFromCert(&cert), knownKeyFn)
```

Comment:
nit: maybe the api `NewServer` should also check if knownKeyFn is nil and returns an error?